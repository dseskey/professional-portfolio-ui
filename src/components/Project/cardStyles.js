export default {
    cardHeaderStyle: { color: 'red', fontWeight:'800', textAlign: 'center', },
    cardContainerStyle: { backgroundSize: 'cover', height: '25em', marginTop: '2em', marginBottom: '2em', marginLeft: '1em', marginRight:'1em' },
    cardContainerStyleHover: {
        background: "grey", height: '20em', marginTop: '1em',
        width: '90%',
        marginLeft: '5%',
        borderRadius: '50px',
        borderStyle: 'outset'
    },
    cardProjectDescriptionStyle: { color: 'black', fontSize: '.7em' }
}