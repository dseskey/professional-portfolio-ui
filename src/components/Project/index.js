import React from 'react';
import { Container, Row, Col, Card, ListGroup, ListGroupItem } from 'react-bootstrap';
import projectJson from '../../assets/json/projects.json';
function Project() {
    return (
        <Container fluid data-testid='project-container' id='project-container' className='pagesParentContainer'>
            <Row className="pageHeader">
                <h2 data-testid='project-section-header'>My Projects</h2>
            </Row>
            <Row>
                {projectJson.projects.map((project) => {
                    return (
                        <Col data-testid={`${project.title}-col`} key={project.title + '-col'}>
                            <Card data-testid={`${project.title}-card`} key={project.title + '-card'} style={{ width: '18rem' }}>
                                {project.imageSRC &&
                                    <Card.Img  data-testid={`${project.title}-card-image`} key={project.title + '-card-image'} variant="top" src={project.imageSRC} />
                                }
                                <Card.Body key={project.title + '-card-body-content'}>
                                    {project.title &&
                                        <Card.Title data-testid={`${project.title}-card-title`} key={project.title + '-card-title'}>{project.title}</Card.Title>
                                    }
                                    {project.description &&
                                        <Card.Text data-testid={`${project.title}-card-text`} key={project.title + '-col'}>
                                            {project.description}
                                        </Card.Text>

                                    }
                                </Card.Body>
                                {/* <ListGroup data-testid={`${project.title}-tech-list`} key={project.title + '-tech-list'} className="list-group-flush">
                                    <ListGroupItem>Cras justo odio</ListGroupItem>
                                    <ListGroupItem>Dapibus ac facilisis in</ListGroupItem>
                                    <ListGroupItem>Vestibulum at eros</ListGroupItem>
                                </ListGroup> */}
                                <Card.Body data-testid={`${project.title}-card-body-links`} key={project.title + '-card-body-links'}>
                                    {project.githubURL &&
                                        <Card.Link href={project.githubURL}>Github</Card.Link>
                                    }
                                    {project.appURL &&
                                        <Card.Link href={project.appURL}>Application Page</Card.Link>
                                    }
                                </Card.Body>
                            </Card>
                        </Col>
                    )
                })}
            </Row>

        </Container>
    )
}

export default Project;