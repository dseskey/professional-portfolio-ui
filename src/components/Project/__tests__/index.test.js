import React from 'react';
import {render, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Project from '..'

afterEach(cleanup);

const project = {
    title: "Travelers Mobile",
    description: "The Travelers mobile application is a React Native based application leveraging AWS and on-prem infrastructure.",
    urls: [
        {
            type: "github",
            url:"http://www.github.com"
        },
        {
            type: "itunes",
            url: "https://apps.apple.com/us/app/travelers-mobile/id354604876#?platform=iphone"
        }
    ],
    imageSRC:"https://s3.us-east-2.amazonaws.com/dseskey.com/images/professional-portfolio/trvMobileApp.jpeg"
}

describe("Project component", () =>{
    it('renders', () => {
        render(<Project project={project}/>);
    });
    it('matches snapshot DOM node structure', () => {
        const {asFragment} = render(<Project project={project}/>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('has a Col container with an ID of the Travelers Mobile-col and class col-md-4', () => {
       const {getByTestId} = render(<Project project={project}/>);
       expect(getByTestId('Travelers Mobile-col')).toHaveClass('col-md-4');

    });

    it('has a Card with a project name of Travelers mobile, class of card, and a style with a background using the S3 URL', () => {
        const {getByTestId} = render(<Project project={project}/>);
        expect(getByTestId('Travelers Mobile-card')).toHaveClass('card');
        expect(getByTestId('Travelers Mobile-card')).toHaveAttribute('data-projectname','Travelers Mobile');
        expect(getByTestId('Travelers Mobile-card')).toHaveStyle('background-image: url(https://s3.us-east-2.amazonaws.com/dseskey.com/images/professional-portfolio/trvMobileApp.jpeg)')
 
     });

     /*TODO: Add a test to validate hover render*/
});

