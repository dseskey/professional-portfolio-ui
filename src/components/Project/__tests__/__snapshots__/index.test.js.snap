// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div
    class="col-md-4"
    data-testid="Travelers Mobile-col"
  >
    <div
      class="card"
      data-projectname="Travelers Mobile"
      data-testid="Travelers Mobile-card"
      style="background-image: url(https://s3.us-east-2.amazonaws.com/dseskey.com/images/professional-portfolio/trvMobileApp.jpeg);"
    />
  </div>
</DocumentFragment>
`;
